packages/core/EventManager.ts

```diff
@@ -179,7 +179,7 @@ export default class EventManager {
       }
       const isCalendarType = isCalendarResult(result);
       if (isCalendarType) {
-        evt.iCalUID = result.iCalUID || undefined;
+        evt.iCalUID = result.iCalUID || event.iCalUID || undefined;
```

Comment:
What's the reasoning behind this logic? 
I am not a fan of adding fallbacks like this as in some scenarios where we expect iCalUID to to be returned(and if it doesn't return) it can wrongly do the fallback. I would prefer to fail in such cases so that the issue can be detected.

1. When do we expect result.iCalUID to be set? Should it be set by a calendar app? If yes, may be we can error if it's not set by Calendar App.
2. What is the scenario when we expect result.iCalUID to be not set but event.iCalUID would be set?
3. What is the scenario when both of them will not be set for a calendar app? Shouldn't we error out in that case?

Reply:
If we know that a particular scenario shouldn't happen but might be happening in production, it might be better to log it as an error so that we can investigate how many times it's happening and in which case exactly.

Reply:
In that case I think we should have a clear way to configure what such apps are and we can do `doesCalendarSupportStoringCustomCalUid()`. So, if a calendar supports that, we  don't use result.iCalUID and if it does we use that. That makes the logic very clear and it would be easier to discover bugs. 

I dread a scenario when iCalUID is still being passed for Google Calendar but it isn't what we expect, so we should have strict checks everywhere.